src/server/es/index.js

```diff
@@ -53,14 +53,21 @@ class ES {
       },
     };
     validatedQueryBody.track_total_hits = true;
-    log.info('[ES.query] index, type, query body: ', esIndex, esType, JSON.stringify(validatedQueryBody));
-    return this.client.search({
+
+    var start = new Date().getTime();
+    var searchResult = this.client.search({
       index: esIndex,
       body: validatedQueryBody,
     }).then((resp) => resp.body, (err) => {
       log.error(`[ES.query] error during querying: ${err.message}`);
       throw new Error(err.message);
-    });
+    }).finally(() => {
+       var end = new Date().getTime();
+       var durationInMS = end - start;
+
+       log.info('[ES.query] DurationInMS:' + durationInMS + '. index, type, query body: ', esIndex, esType, JSON.stringify(validatedQueryBody));
+       return searchResult;
```

Comment:
Don't return from the finally - you can actually get rid of searchResult assignment and `return this.client.search(...)` like it was before. This function needs to return the Promise for the async call so callers of the functions can do their own async response handling.

Reply:
:+1: I'll push and then I'm going to test in my dev env